src/main/webapp/WEB-INF/view/chat.jsp

```diff
@@ -90,6 +92,59 @@ List<Message> messages = (List<Message>) request.getAttribute("messages");
     <% if (request.getSession().getAttribute("user") != null) { %>
     <form action="/chat/<%= conversation.getTitle() %>" method="POST">
         <input type="text" name="message">
+        
+        <!-- Button trigger modal -->
+		<button type="button" class="btn btn-primary" data-toggle="modal" data-target="#exampleModalLong">
+		  Send a GIF
+		</button>
+		
+		<!-- Modal -->
+		<div class="modal fade" id="exampleModalLong" tabindex="-1" role="dialog" aria-labelledby="exampleModalLongTitle" aria-hidden="true">
+		  <div class="modal-dialog" role="document">
+		    <div class="modal-content">
+		      <div class="modal-header">
+		        <input type="search" class="form-control" id="GIFsearch" placeholder="Search GIFs on GIPHY...">
+		        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+		          <span aria-hidden="true">&times;</span>
```

Comment:
I had to look this up to realize it's inserting a multiplication sign as the close button X! That's a neat styling trick.